src/index.ts

```diff
@@ -0,0 +1,257 @@
+    // validation here.
+    if (record.httpRequest) {
+      entryMetadata.httpRequest = record.httpRequest;
+      delete record.httpRequest;
+    }
+
+    // record does not have index signature.
+    // tslint:disable-next-line:no-any
+    if ((record as any)[LOGGING_TRACE_KEY]) {
+      // tslint:disable-next-line:no-any
+      entryMetadata.trace = (record as any)[LOGGING_TRACE_KEY];
+      // tslint:disable-next-line:no-any
+      delete (record as any)[LOGGING_TRACE_KEY];
+    }
+
+    return this.stackdriverLog.entry(entryMetadata, record);
+  }
+
+  /**
+   * Intercept log entries as they are written so we can attempt to add the
+   * trace ID in the same continuation as the function that wrote the log,
+   * because the trace agent currently uses continuation local storage for the
+   * trace context.
+   *
+   * By the time the Writable stream buffer gets flushed and _write gets called
+   * we may well be in a different continuation.
+   */
+  // Writable.write used 'any' in function signature.
+  // tslint:disable-next-line:no-any
+  write(record: any, callback?: Function): boolean;
+  // tslint:disable-next-line:no-any
+  write(record: any, encoding?: string, callback?: Function): boolean;
+  // tslint:disable-next-line:no-any
+  write(...args: any[]): boolean {
+    let record = args[0];
+    let encoding: string|null = null;
+    let callback: Function;
+    if (typeof args[1] === 'string') {
+      encoding = args[1];
+      callback = args[2];
+    } else {
+      callback = args[1];
+    }
+    record = extend({}, record);
+    // record does not have index signature.
+    // tslint:disable-next-line:no-any
+    if (!(record as any)[LOGGING_TRACE_KEY]) {
+      const trace = getCurrentTraceFromAgent();
+      if (trace) {
+        // tslint:disable-next-line:no-any
+        (record as any)[LOGGING_TRACE_KEY] = trace;
+      }
+    }
+    if (encoding !== null) {
+      return super.write.call(this, record, encoding, callback);
+    } else {
+      return super.write.call(this, record, callback);
+    }
+  }
+
+  /**
+   * Relay a log entry to the logging agent. This is called by bunyan through
+   * Writable#write.
+   *
+   * @param {object} record - Bunyan log record.
+   *
+   * @private
+   */
+  _write(record: BunyanLogRecord, encoding?: string, callback?: Function) {
+    const entry = this.formatEntry_(record);
+    this.stackdriverLog.write(entry, callback);
+  }
+
+  /**
+   * Relay an array of log entries to the logging agent. This is called by
+   * bunyan through Writable#write.
+   *
+   * @param {object[]} records - Array of WritableStream WriteReq objects.
+   *
+   * @private
+   */
+  // Writable._writev used 'any' in function signature.
+  _writev(
+      chunks: Array<{
+        // tslint:disable-next-line:no-any
+        chunk: any;
+        encoding: string;
+      }>,
+      callback?: Function) {
+    // tslint:disable-next-line:no-any
+    const entries = chunks.map((request: any) => {
```

Comment:
Seems like we at least know `request` has a `chunk` property.  Would be nice to have better types.

Reply:
Replace `any` to type 
```
{
     // tslint:disable-next-line:no-any
     chunk: any;
     encoding: string;
 }
```
chunk is in `any` type because Writable.write uses `any`